### PR TITLE
Avoid calculating mutual information on unique columns

### LINKED ISSUE #555: Prevent Mutual Info from being calculated on unique columns
- If a column is unique, mutual info should not be calculated for that column.

### PR DESCRIPTION
**Pull request by tamargrey**
-  Closes #555 
- Had to update some of the existing tests to not be unique

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -77,6 +77,12 @@ jobs:
           ref = basename("${{ github.event.ref }}")
           is_dev = not any(pattern.match(ref) for pattern in patterns)
           print('::set-output name=is_dev::' + str(is_dev))
+      - if: ${{ steps.branch.outputs.is_dev == 'True' }}	
+        name: Checkout repository	
+        uses: actions/checkout@v2	
+      - if: ${{ steps.branch.outputs.is_dev == 'True' }}	
+        name: Check if release notes were updated	
+        run: cat docs/source/release_notes.rst | grep ":pr:\`${{ steps.pull_request.outputs.number }}\`"
```
Comment (tamargrey) [inline .github/workflows/tests.yml:85]: Added these lines back in to get the release note update CI check working again

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -77,7 +77,15 @@ jobs:
...
-
+      - if: ${{ steps.branch.outputs.is_dev == 'True' }}
+        name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - if: ${{ steps.branch.outputs.is_dev == 'True' }}
+        name: Check if release notes were updated
+        run: cat docs/source/release_notes.rst | grep ":pr:\`${{ github.event.number }}\`"
```
Comment (tamargrey) [inline .github/workflows/tests.yml:88]: Added these lines back in to get the release note update CI check working again

File: woodwork/datatable.py [hunk 3]
```diff
@@ -1023,6 +1023,9 @@ def mutual_information_dict(self, num_bins=10, nrows=None):
         if ks and isinstance(self._dataframe, ks.DataFrame):
             data = data.to_pandas()
 
+        # Dpn't calculate mutual info for columns that aren't unique
+        data = data[[col for col in data.columns if not data[col].is_unique]]
```
Comment (gsheni) [inline woodwork/datatable.py:1027]: Can we move the following line out `_handle_nans_for_mutual_info`, and do the following:
```python
        if nrows is not None and nrows < data.shape[0]:
            data = data.sample(nrows)

        # Remove fully null columns
        data = data.loc[:, data.columns[data.notnull().any()]]
        # Remove columns that are unique
        data = data[[col for col in data.columns if not data[col].is_unique]]

        data = self._handle_nans_for_mutual_info(data)
        data = self._make_categorical_for_mutual_info(data, num_bins)
```
- My thought is sample will do the most with large datasets to reduce the size. 
  - After that columns that are all NaN can be removed, and then any that are unique. (The unique column could make for a large computation...)
- Maybe we rename `_handle_nans_for_mutual_info` to `_replace_nans_for_mutual_info`?

Comment (tamargrey) [reply woodwork/datatable.py:1027]: Yeah, I like this. It also probably makes the behavior of mutual info/what it actually does to the data prior to calculating more clear

Comment (tamargrey) [reply woodwork/datatable.py:1027]: Question: is it faster to only create the new DataFrames from removing fully null columns and unique columns if there are any columns to remove? 

Right now, if there's nothing to remove we're needlessly copying the DataFrame

Comment (gsheni) [reply woodwork/datatable.py]: @tamargrey good point. Yes, we should only do this if there is fully NULL columns and/or unique columns.

Comment (tamargrey) [reply woodwork/datatable.py]: ok, updated to only do this when necessary

**Review by gsheni — APPROVED** [review]
lgtm

**Pull request merged**